packages/neon-api/src/transaction/signer.ts

```diff
@@ -0,0 +1,105 @@
+import { tx, wallet, u } from "@cityofzion/neon-core";
+
+/**
+ * A class with functions to sign transaction
+ */
+export class TransactionSigner {
+  /**
+   * transaction to be signed
+   */
+  public transaction: tx.Transaction;
+
+  constructor(transaction: tx.Transaction) {
+    this.transaction = transaction;
+  }
+
+  /**
+   * Sign a transaction with Accounts or Private Keys
+   * @description This is used when you have full access to signer accounts
+   * @param accounts accounts that will sign this transaction
+   */
+  public signWithAccount(...accounts: (wallet.Account | string)[]) {
+    accounts.forEach(account => {
+      this._checkAcc(account);
+      this.transaction.sign(account);
+    });
+  }
+
+  /**
+   * Sign a transaction with Witnesses
+   * @description This can be used when you accept a signature from someone else
+   * @param witnesses witnesses that will be added to the transaction
+   */
+  public signWithWitness(...witnesses: tx.Witness[]) {
+    witnesses.forEach(witness => {
+      this._checkWitness(witness);
+      this.transaction.addWitness(witness);
+    });
+  }
+
+  /**
+   * Sign a transaction with multi-signatures for multi-sig account
+   * @param multisigAccount multisig account
+   * @param witnesses signatures from accounts within the multisig-account
+   */
+  public signWithMultiSigAccount(
+    multisigAccount: wallet.Account,
+    ...witnesses: tx.Witness[]
+  ) {
+    this._checkMultisigAcc(multisigAccount);
+    const multisigWitness = tx.Witness.buildMultiSig(
+      this.transaction.serialize(),
+      witnesses,
+      multisigAccount
+    );
+    this.transaction.addWitness(multisigWitness);
+  }
+
+  private _checkAcc(account: wallet.Account | string) {
+    const acc = new wallet.Account(account);
+    this._checkScripthash(acc.scriptHash);
+  }
+
+  private _checkWitness(witness: tx.Witness) {
+    const publicKeys = wallet.getPublicKeysFromVerificationScript(
```

Comment:
I dont understand all this. Why go through so much when you can calculate the scripthash from the verification script directly in one step? Its the same scenario for both single and multisig accounts.